tp/tp4-ingress.md

```diff
@@ -0,0 +1,55 @@
+## Introduction
+
+In the previous exercise, you have exposed the *whoami* application internally using a service of type ClusterIP.
+Now you want to go public and open your application to external clients. One way to do this is by using a Kubernetes Ingress.
+
+Ingress exposes HTTP and HTTPS routes from outside the cluster to services within the cluster.
+To be able to create Ingress objects in your cluster, you need to first install an *Ingress Controller*. 
+We will use traefik, a powerful Ingress Controller and reverse proxy (https://containo.us/traefik/)
+
+## Install traefik
+
+Enable the RBAC:
+```shell script
+kubectl apply -f tp4-traefik-rbac.yml
+```
+
+Install *traefik*:
+```shell script
+kubectl apply -f tp4-treafik-ingress-controller.yml
+```
+
+## Expose the whoami application using an Ingress
+
+Traefik is deployed as a Deployment with one replica. Find out the node on which runs the corresponding Pod:
+```shell script
+kubectl get pod -o wide -n kube-system | grep traefik
+``` 
```

Comment:
Traefik as a deploiment (node can change) or a daemonset (only need to get one ip/dns of one node but it's install on all node) ?

Reply:
Je voulais pas les embrouiller avec ce niveau de détail :)

Reply:
:+1: tu as raison